examples/test_util/EclIO.cpp

```diff
@@ -0,0 +1,808 @@
+    std::reverse(floatToConvert, floatToConvert+4);
+    
+    return value;
+}
+
+template<>
+auto EclIO::flipEndian<double>(const double &num) const {
+    
+    double value=num;
+    
+    char *doubleToConvert = reinterpret_cast<char*>(&value);
+    std::reverse(doubleToConvert, doubleToConvert+8);
+
+    return value;
+}
+
+std::tuple<const int, const int> EclIO::block_size_data(EIOD::eclArrType arrType){
+
+    switch(arrType) {
+        case EIOD::INTE  : return std::make_tuple(EIOD::sizeOfInte,EIOD::MaxBlockSizeInte);   break;
+        case EIOD::REAL  : return std::make_tuple(EIOD::sizeOfReal,EIOD::MaxBlockSizeReal);   break;
+        case EIOD::DOUB  : return std::make_tuple(EIOD::sizeOfDoub,EIOD::MaxBlockSizeDoub);   break;
+        case EIOD::LOGI  : return std::make_tuple(EIOD::sizeOfLogi,EIOD::MaxBlockSizeLogi);   break;
+        case EIOD::CHAR  : return std::make_tuple(EIOD::sizeOfChar,EIOD::MaxBlockSizeChar);   break;
+        case EIOD::MESS  : throw std::invalid_argument("Type 'MESS' have not assosiated data") ;   break;
+    }
+}
+
+std::string EclIO::trimr(const std::string &str1) const {
+
+    std::string tmpStr=str1;
+    int l=tmpStr.length()-1;
+    
+    while ((tmpStr[l]==' ') && (l>=0))
+        l--;
+
+    return tmpStr.substr(0,l+1);
+    
+}
+
+
+void EclIO::readBinaryHeader(std::fstream &fileH,std::string &arrName,int &size,EIOD::eclArrType &arrType){
+
+    int bhead;
+    std::string tmpStrName(8,' ');
+    std::string tmpStrType(4,' ');
+
+    if (!fileH.is_open()){
+        std::string message="fstream fileH not open for reading";
+        throw std::runtime_error(message);
+    }
+
+    fileH.read((char *)&bhead, sizeof(bhead));
+    bhead=flipEndian<int>(bhead);
+    
+    if (bhead!=16){
+        std::string message="Error reading binary header. Expected 16 bytes of header data, found " + std::to_string(bhead);
+        throw std::runtime_error(message);
+    }
+    
+    fileH.read(&tmpStrName[0], 8);
+    
+    fileH.read((char *)&size, sizeof(size));
+    size=flipEndian<int>(size);
+
+    fileH.read(&tmpStrType[0], 4);
+
+    fileH.read((char *)&bhead, sizeof(bhead));
+    bhead=flipEndian<int>(bhead);
+    
+    if (bhead!=16){
+        std::string message="Error reading binary header. Expected 16 bytes of header data, found " + std::to_string(bhead);
+        throw std::runtime_error(message);
+    }
+    
+    arrName=tmpStrName;
+    
+    if (tmpStrType=="INTE")
+        arrType=EIOD::INTE;
+    else if (tmpStrType=="REAL")    
+        arrType=EIOD::REAL;
+    else if (tmpStrType=="DOUB")    
+        arrType=EIOD::DOUB;
+    else if (tmpStrType=="CHAR")    
+        arrType=EIOD::CHAR;
+    else if (tmpStrType=="LOGI")    
+        arrType=EIOD::LOGI;
+    else
+        throw std::runtime_error("Error, unknown array type '" + tmpStrType +"'");
+    
+}
+
+unsigned long int EclIO::calcSizeOfArray(const int num, const EIOD::eclArrType &arrType){
```

Comment:
This is the number of bytes occupied by the keyword on disk, need a better name.

Reply:
I am having problems coming up with a better name, maybe you have a suggestion? Can you also elaborate a bit on what you mean by generic implementation?

Reply:
Generic: one common implementation for all the datatypes.

Reply:
Name: sizeOnDisk?